### DISCUSSION TITLE
Best way to use a ruby gem as a build dependency?

### DISCUSSION THREAD

**Discussion by mhanberg** [Homebrew/discussions#2873] [Writing Formulae/Casks] [answered]
Hi!

I am writing my own "personal cli" and am going to host my own tap/formula to make it easy to install.

I am using a ruby gem called [bashly](https://github.com/DannyBen/bashly) to generate the final artifact (a single bash script) and I am unsure on the best way to install the gem as only a build dependency.

I have looked at some examples of using gems in Formula, but most of them are about using brew to install the gem itself.([solargraph](https://github.com/Homebrew/homebrew-core/blob/e0902a367f427c3729ced06fdde17220cc2947cb/Formula/solargraph.rb) for example).

Any examples  that I should check out?

Thanks!

**Comment by cho-m** [comment, marked as answer]
The closest examples I can think of are some Formulae that use a specific `bundler` like:
- [`swiftgen`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/swiftgen.rb#L27-L29)
  ```rb
      ENV["GEM_HOME"] = buildpath/"gem_home"
      system "gem", "install", "bundler"
      ENV.prepend_path "PATH", buildpath/"gem_home/bin"
  ```
- [`mailcatcher`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/mailcatcher.rb#L97-L99)
  ```rb
      ENV["GEM_HOME"] = buildpath/"gem_home"
      system "gem", "install", "--no-document", resource("bundler").cached_download
      with_env(PATH: "#{buildpath}/gem_home/bin:#{ENV["PATH"]}") do
  ```

---

In general, the idea would be to use a directory under `buildpath` since this is a temporary location where the build will take place.

In above examples, the 1st downloads an unversioned/latest `bundler` while the 2nd downloads a specific version of `bundler` (v2.2.25) listed as resource, which would help make builds more reproducible.

**Reply by mhanberg** [reply]
This worked, thanks!
